internal/k8s/controller.go

```diff
@@ -914,6 +929,8 @@ func (lbc *LoadBalancerController) syncVirtualServer(task task) {
 		lbc.recorder.Eventf(vs, api_v1.EventTypeWarning, "IgnoredVirtualServerRoute", "Ignored VirtualServerRoute %v: %v", vsrError.VirtualServerRouteNsName, vsrError.Error)
 		if vsrError.VirtualServerRoute != nil {
 			lbc.recorder.Eventf(vsrError.VirtualServerRoute, api_v1.EventTypeWarning, "Ignored", "Ignored by VirtualServer %v/%v: %v", vs.Namespace, vs.Name, vsrError.Error)
+			vsrKey := fmt.Sprintf("%v/%v", vsrError.VirtualServerRoute.Namespace, vsrError.VirtualServerRoute.Name)
+			vsrs = removeVirtualServerRouteByKey(vsrKey, vsrs)
```

Comment:
the function `removeVirtualServerRouteByKey` modifies the array that back ups the slice `vsrs`. So it will modify the array of the slices that configurator stores. That could lead to hard to catch bugs :(

perhaps instead of calling `removeVirtualServerRouteByKey` here and in line 998, you have another slice, something like 

```go
var handledVSRs []*conf_v1.VirtualServerRoute
. . .
handledVSRs = append(handledVSRs, vsrError.VirtualServerRoute)
. . .
handledVSRs = append(handledVSRs, vsr)
. . .
ignoredVSRs := filterOutVirtualServerRoutes(vsrs, handledVSRs)
for _, vsr := range ignoredVSRs 
. . .
```
In `filterOutVirtualServerRoutes`, make sure to create a new slice, instead of modifying the existing.

I would recommend renaming `vsrs` to `previosVSRs` to reflect that the slice contains a list of the VSRs that were previously added in the configuration.